desktop/src/pages/ChatPage/ChatInput/index.tsx

```diff
@@ -74,89 +136,105 @@ export default function ChatInput(_props: ChatInputProps) {
         const textarea = e.currentTarget;
         const start = textarea.selectionStart;
         const end = textarea.selectionEnd;
-        const newMessage = message.substring(0, start) + '\n' + message.substring(end);
-        setMessage(newMessage);
+        const newValue = input.substring(0, start) + '\n' + input.substring(end);
 
+        setInput(newValue);
+
+        // Move cursor position after the new line
         setTimeout(() => {
-          textarea.setSelectionRange(start + 1, start + 1);
+          textarea.selectionStart = textarea.selectionEnd = start + 1;
         }, 0);
-      } else {
+      } else if (!e.shiftKey) {
         e.preventDefault();
-        handleSubmit();
+        if (!disabled && canSend) {
+          onSubmit();
+        }
       }
     }
   };
 
   const handleModelChange = (modelName: string) => {
     setSelectedModel(modelName);
+    setMessages([]); // Clear chat history when changing models
```

Comment:
Any reason to clear chat history when changing model? Users can be upset if they loose it